File: Python/BEGINNER/3174 Noel's Work Groups.py
# SUBPROGRAM
def PresentProduction(numElves):
    sumDolls = sumArchitects = sumMusicians = sumDesigners = 0
    for i in range(numElves):
        name, profession, workTime = input().split()
        if profession == "bonecos":
            sumDolls += int(workTime)
        elif profession == "arquitetos":
            sumArchitects += int(workTime)
        elif profession == "musicos":
            sumMusicians += int(workTime)
        else:
            sumDesigners += int(workTime)
    print("%d" % ((sumDolls//8) + (sumArchitects//4) + (sumMusicians//6) + (sumDesigners//12)))
    return None

# MAIN
numberOfElves = int(input())
PresentProduction(numberOfElves)

File: C/BEGINNER/3174 Noel's Work Groups.c
#include <stdio.h>
#include <string.h>

void produceGifts(int numberOfElves);

int main()
{
    int numberOfElves;
    scanf("%d", &numberOfElves);
    produceGifts(numberOfElves);
    return 0;
}

void produceGifts(int numElves)
{
    char name[100], profession[100];
    int workTime, sumDolls=0, sumArchitects=0, sumMusicians=0, sumDesigners=0;
    for (int i=0; i < numElves; i++)
    {
        scanf("%s %s %d", name, profession, &workTime);
        if (strcmp(profession, "bonecos") == 0)
            sumDolls += workTime;
        else if (strcmp(profession, "arquitetos") == 0)
            sumArchitects += workTime;
        else if (strcmp(profession, "musicos") == 0)
            sumMusicians += workTime;
        else
            sumDesigners += workTime;
    }
    printf("%d\n", ((sumDolls/8) + (sumArchitects/4) + (sumMusicians/6) + (sumDesigners/12)));
}
